tuna/rocmlir/rocmlir_tables.py

```diff
@@ -0,0 +1,296 @@
+  """Collects the results of convolution tuning.
+  """
+  __tablename__ = "rocmlir_conv_results"
+  __table_args__ = (UniqueConstraint("config", "session", name="uq_idx"),)
+
+  @orm.reconstructor
+  def __init__(self, **kwargs):
+    if 'logger' in kwargs:
+      self.logger = kwargs['logger']
+    else:
+      self.logger = setup_logger('results')
+
+  @declared_attr
+  def session(self):
+    """session column"""
+    return Column(Integer, ForeignKey("session_rocmlir.id"), nullable=False)
+
+  config = Column(Integer, ForeignKey("rocmlir_conv_config.id"), nullable=False)
+  config_str = Column(Text, nullable=False)
+
+  perf_config = Column(Text, nullable=False)
+  kernel_tflops = Column(Float, nullable=False)
+
+  def get_query(self, sess, result_obj, session_id):
+    """Construct a Db query for the find object
+    """
+    query = sess.query(result_obj).filter(result_obj.session == session_id,
+                                          result_obj.config == self.config)
+    self.logger.info("result query %s-%s", session_id, self.config)
+    return query
+
+  # +++pf:  rewrite me for tuningRunner.py output!
+  def parse(self, lines):
+    """parse logger output line for result data """
+    line = lines.splitlines()[-1]
+    print(f"line being parsed is '{line}'", file=sys.stderr)
+    return line.split('\t')
+
+
+#pylint: disable=too-few-public-methods
+class RocMLIRDBTables(DBTablesInterface):
+  """Represents db tables for rocMLIR lib"""
+
+  def __init__(self, **kwargs):
+    """Constructor"""
+    super().__init__(**kwargs)
+    allowed_keys = set(['config_type', 'session_id'])
+
+    self.config_type = None
+    self.job_table = None
+    self.session_table = SessionRocMLIR
+    self.config_table = None
+    self.results = None
+
+    self.__dict__.update(
+        (key, value) for key, value in kwargs.items() if key in allowed_keys)
+    self.set_tables()
+
+  def set_tables(self, sess_class=SessionRocMLIR):
+    """Set appropriate tables based on requirements"""
+    super().set_tables(sess_class)
+    # +++pf:  branch on self.config_type when we have GEMM, too.
+    self.job_table = ConvolutionJob
+    self.config_table = ConvolutionConfig
+    self.results = ConvolutionResults
+
+
+def get_tables() -> List[BASE]:
+  """Returns a list of all Example lib DB tables"""
+  tables: List[BASE] = []
+  with DbSession() as session:
+    engine = session.bind
+    connect = session.connection()
+
+    def append_if_not_exists(table):
+      # Note: this changes in sqlalchemy 1.4.
+      if not inspect(engine).dialect.has_table(connect, table.__tablename__):
+        tables.append(table)
+
+    append_if_not_exists(SessionRocMLIR())
+    append_if_not_exists(Machine(local_machine=True))
+    append_if_not_exists(ConvolutionConfig())
+    append_if_not_exists(ConvolutionJob())
+    append_if_not_exists(ConvolutionResults())
+
+  return tables
+
+
+def clear_tables():
```

Comment:
Also for testing, ensure that the tables are in a known empty state.  Used in some tests/test*rocmlir.py files.